app/models/search_result.rb

```diff
@@ -0,0 +1,39 @@
+class SearchResult
+  include NameConstructable
+  include SubmissionTagHelper
+
+  def initialize(data)
+    @data = data
+  end
+
+  delegate :id, to: :submission
+
+  def submission
+    @submission ||= Submission.find_by(id: @data[:application_id]) || UpdateSubmission.call(@data.deep_stringify_keys)
+  end
+
+  def laa_reference
+    submission.data['laa_reference']
+  end
+
+  def firm_name
+    submission.data['firm_name']
```

Comment:
any reason we can't look this up form the response JSON? try and avoid touching the local DB as much as possible?